sdk/storage/Azure.Storage.DataMovement/src/Shared/DataMovementExtensions.cs

```diff
@@ -336,5 +306,8 @@ internal static void VerifyJobPartPlanHeader(this JobPartInternal jobPart, JobPa
                 throw Errors.MismatchResumeCreateMode(header.ForceWrite, jobPart._createMode);
             }
         }
+
+        internal static bool IsLocalResource(this StorageResource resource)
```

Comment:
I have high doubts of a local resource type being added outside the base package.

This was actually the suggestion from our archboard meeting to do.

Using the Uri itself to determine what's a "local" resource and not then we have to parse the Uri, which does take longer I feel like. Also I worry about checking the `Uri` since sometimes the `Uri` can have it's own format. I think this would be easy to update in the future we add some other "Local" Resource